tools/wasm_demo/README.md

```diff
@@ -3,6 +3,43 @@
 This folder contains an example how to decode JPEG XL files on a web page using
 WASM engine.
 
+### One line demo
+
+The simplest way to get support of JXL images on client side is simply link
+one extra script (`<script src="service_worker.js">`) to the page. This script
+installs a `ServiceWorker` that:
+
+ - checks is the browser supports JXL image format already
+ - if it is not, then advertise `image/jxl` media format in image requests
+ - then, if server responds with `image/jxl` content it gets decoded and
+   re-encoded to PNG on the flight
+
+Generally the message / data flow looks the following way:
+
+ - `Fetch API` receives resource request from client page (e.g. when HTML
+   engine discovers `img` tag) and asks `ServiceWorker` how to proceed
+ - `ServiceWorker` alters the request and uses `Fetch API` to obtain data
+ - when data arrives, `ServiceWorker` forwards it to "client" (the page)
+   that initiated the resource request
+ - client forwards data to a worker (see `client_worker.js`) to avoid
+   processing in the "main loop" thread
+ - worker does actual decoding; to make it faster several additional
+   workers are spawned (to enable multi-threading in WASM module);
+   decoded image is wrapped in non-compressed PNG format and sent back
+   to client
+ - client relays image data to `ServiceWorker`
+ - `ServiceWorker` passes data to `Fetch API` as a response to initial
+   resource request
```

Comment:
```suggestion
   engine discovers an `img` tag) and asks the `ServiceWorker` how to proceed
 - the `ServiceWorker` alters the request and uses the `Fetch API` to obtain data
 - when data arrives, the `ServiceWorker` forwards it to the "client" (the page)
   that initiated the resource request
 - the client forwards the data to a worker (see `client_worker.js`) to avoid
   processing in the "main loop" thread
 - a worker does the actual decoding; to make it faster several additional
   workers are spawned (to enable multi-threading in WASM module);
   the decoded image is wrapped in non-compressed PNG format and sent back
   to client
 - the client relays image data to `ServiceWorker`
 - the `ServiceWorker` passes data to `Fetch API` as a response to initial
   resource request
```